### PR TITLE
Eliminate recursion from the internals of term formula removal

### PR DESCRIPTION
**Pull request by ajreynol**
This makes all recursion (applying term formula removal on lemmas introduced by term formula removal) optionally happen at the top level call.

This is in preparation for making term formula removal lazier, in which case we will only apply one step of term formula removal at a time.

One QF_UFNIA regression times out due to changing the search, an option is changed for this benchmark.

### CODE REVIEW

File: src/smt/term_formula_removal.cpp [hunk 1]
```diff
@@ -50,14 +50,77 @@ RemoveTermFormulas::~RemoveTermFormulas() {}
...
+  {
+    return theory::TrustNode::null();
+  }
+  // if running to fixed point, we run each new assertion through the
+  // run lemma method
+  if (fixedPoint)
+  {
+    size_t i = 0;
+    std::unordered_set<Node, NodeHashFunction> processed;
+    while (i < newAsserts.size())
```
Comment (HanielB) [inline src/smt/term_formula_removal.cpp:68]: why not a `for`?

**Review by HanielB — APPROVED** [review]
LGTM, just a minor comment

File: src/smt/term_formula_removal.cpp (hunk 1 above)
Comment (ajreynol) [reply src/smt/term_formula_removal.cpp:68]: Because newAsserts is being added to in the loop.

Comment (HanielB) [reply src/smt/term_formula_removal.cpp:68]: So the size will change? Having the condition for the `for` as you have for the `while` would amount to the same, no?

Comment (ajreynol) [reply src/smt/term_formula_removal.cpp:68]: i think its clearer this way, since it suggests that a fixed point is being run.

**Pull request merged**